### ISSUE TITLE
Group articulations won't form on some occasions

### ISSUE THREAD

**Issue by CTegtmeyer** [spawaskar-cora/cora-docs#392]
Every so often, a group articulation, will not form, even after repeated attempts at entering it. This has happened with hand and foot group creations. The groups I form after create with no problems, but these groups just never appear?

**Comment by SachinPawaskarUNO** [comment]
@CTegtmeyer, This was a difficult one to identify, I think I have narrowed these down to timeout issues due to resources. Currently we have some of these specimen group creation jobs taking longer than the allowed job time of 60 seconds on the current resources. For now I'm going to increase the allowed job time to 90 seconds. We can fine tune this timeout/allowed timeout value as we continue to use the specimen group creation feature and in the future when we have additional resources. @CTegtmeyer My guess is that this is happening for groups that have quite a few bones in them like hand/foot/etc. Can you please confirm?

**Comment by CTegtmeyer** [comment]
Yes, I've only had it happen with the hands and feet, however, they have not been full groups, but have been partial groups, so have not had that many elements in them. I have had vertebrae and rib groups with more elements that have had no problems forming.

**Comment by SachinPawaskarUNO** [comment]
@CTegtmeyer, It essentially has to do with how much work needs to get done in a given Specimen Group Creation job, which is highly variable depending on how many articulations it has to create (hands and feet have a lot more articulations amongst themselves compared to ribs/vertebrae groups which a little less more articulations amongst themselves. Add to this additional associations that may need to be created depending upon whether the user/anthropologist selected adding of traumas/pathology/taphonomy to these bones.
@CTegtmeyer I would like you to keep track of this, as I have made some changes to the timeout values. I'm hoping you still have quite a few specimen bone groups still to create for the Tarawa project. I'm hopeful that the tuning I have done will at least reduce this from happening, if not eliminate this completely.
@CTegtmeyer @fedamann You are getting up there in terms of specimens that have been created in the Tarawa project, more than 10,000, Last count was 10,321. Way to go!!!

**Comment by CTegtmeyer** [comment]
So, it turns out that these groups do form, sometimes hours later. Instead of creating an error when the composite key is already used, it just moved on to the next designator number in the sequence. Is it possible to create a fail-safe that if you accidentally double create a group that it won't form the second group if the composite key is already used?

We still have a lot of specimens to add. So I'll be able to test if it keeps happening.
